From 1093fc1fce0bd03f1768bfe990818ee8d0d0bc7a Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Thu, 13 Jun 2024 12:55:08 +0300
Subject: [PATCH] [FSW-418] fs1: bootrom: reduce tpm timeout values

TPM execution cannot have timeouts as long as
750ms or 2000ms, as these prolonged timeouts
can cause the boot ROM to fail to boot within
the PCIe boot timing restrictions.

This patch reduces timeouts to 2ms or 10ms.

Change-Id: I221d4c32a66d6d945878d6aae309cbac45f876b3
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/common/tpm2.c b/common/tpm2.c
index 5a13366..d5c03e6 100644
--- a/common/tpm2.c
+++ b/common/tpm2.c
@@ -260,7 +260,7 @@
 		if (burstcount)
 			return burstcount;
 
-	} while (!boot_rom_timeout_expired(start, MS_TO_US(TIS_SHORT_TIMEOUT_MS)));
+	} while (!boot_rom_timeout_expired(start, MS_TO_US(TPM_SHORT_TIMEOUT_MS)));
 
 	return -1;
 }
@@ -277,14 +277,14 @@
 
 		if ((*status & mask) == mask)
 			return 0;
-	} while (!boot_rom_timeout_expired(start, MS_TO_US(timeout)));
+	} while (!boot_rom_timeout_expired(start, timeout));
 
 	return -1;
 }
 
 static uint8_t tpm_tis_spi_valid_status(uint8_t *status)
 {
-	return tpm_spi_wait_for_stat(TPM_STS_VALID, TIS_SHORT_TIMEOUT_MS,
+	return tpm_spi_wait_for_stat(TPM_STS_VALID, MS_TO_US(TPM_SHORT_TIMEOUT_MS),
 					status);
 }
 
@@ -296,7 +296,7 @@
 
 	while (size < count &&
 	       tpm_spi_wait_for_stat(TPM_STS_DATA_AVAIL | TPM_STS_VALID,
-					 TIS_SHORT_TIMEOUT_MS, &status) == 0) {
+					 MS_TO_US(TPM_LONG_TIMEOUT_MS), &status) == 0) {
 		burstcnt = tpm_spi_get_burstcount();
 		if (burstcnt < 0)
 			return burstcnt;
@@ -357,7 +357,7 @@
 		if (ret != -ENOENT)
 			return ret;
 
-	} while (!boot_rom_timeout_expired(start, MS_TO_US(TIS_SHORT_TIMEOUT_MS)));
+	} while (!boot_rom_timeout_expired(start, MS_TO_US(TPM_SHORT_TIMEOUT_MS)));
 
 	return ret;
 }
@@ -426,7 +426,7 @@
 			goto out_err;
 
 		ret = tpm_spi_wait_for_stat(TPM_STS_COMMAND_READY,
-						TIS_LONG_TIMEOUT_MS, &status);
+					MS_TO_US(TPM_SHORT_TIMEOUT_MS), &status);
 		if (ret < 0 || !(status & TPM_STS_COMMAND_READY))
 			goto out_err;
 	}
